dotCMS/src/main/java/com/dotcms/rest/ResourceResponse.java

```diff
@@ -112,16 +120,16 @@ public Response response ( String response, CacheControl cacheControl ) {
             }
         }
 
-        Response.ResponseBuilder responseBuilder;
-        if ( contentType != null ) {
-            responseBuilder = Response.ok( response, contentType );
-        } else {
-            /*
-            If the Content type of the response is null the default
-            will be the defined by the @Produces annotation of the RESTful method
-             */
-            responseBuilder = Response.ok( response );
-        }
+        final Response.ResponseBuilder responseBuilder =
+                ( contentType != null )?
+                    Response.ok( response, contentType ):
+                    /*
+                    If the Content type of the response is null the default
+                    will be the defined by the @Produces annotation of the RESTful method
+                     */
+                    Response.ok( response );
+
+        status.ifPresent( theStatus ->  responseBuilder.status(status.get()));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'theStatus' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)